### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by Armetiis**
This PR cleans up some code that caused linter to generate warnings. For more info on the specific changes, refer to the commit titles for each change, as I tried my best to have 1 commit for each warning for ease of review

### CODE REVIEW

File: app/src/main/java/com/github/se/cyrcle/model/address/Address.kt [hunk 1]
```diff
@@ -110,12 +110,12 @@ data class Address(
    * @return the suggestion formatted display name
    */
   fun suggestionFormatDisplayName(maxLength: Int, mode: Mode): String {
-    val fieldPriorities: List<String>
-    if (mode == Mode.LIST) {
-      fieldPriorities = listOf(publicName, road, city, country)
-    } else {
-      fieldPriorities = listOf(publicName, road, city, region, country)
-    }
+    val fieldPriorities: List<String> =
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/model/address/Address.kt:113]: nice change!

File: app/src/main/java/com/github/se/cyrcle/model/parking/Parking.kt [hunk 2]
```diff
@@ -47,6 +47,17 @@ data class Parking(
...
     val tile: Tile = TileUtils.getTileFromPoint(location.center)
 ) {
+
+  /**
+   * Finds a ParkingImage in the imageObjects list that matches the given imagePath.
+   *
+   * @param destinationPath The imagePath to match.
+   * @return The ParkingImage with the matching imagePath, or null if no match is found.
+   */
+  fun findImageByPath(destinationPath: String): ParkingImage? {
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/model/parking/Parking.kt:57]: my bad on this one 😅

**Review by matttsss — APPROVED** [review]
It's all good for me, the app is still well functional and this is a nice cleanup, thanks!

**Review by LouisJamesVasseur — APPROVED** [review]
Every one of the changes you suggest feels reasonable. For this PR since every thing in the code is objectively correct I just checked that the app still works the same and it does!

**Pull request merged**